### PR TITLE
api: add hardcoded versioning support

### PR DESCRIPTION
**Pull request by GRISHNOV**
Add versioning support, now the module api has `_VERSION` hardcoded variable. Is part of the task [1].

1. https://github.com/tarantool/roadmap-internal/issues/204

### CODE REVIEW

File: test/compatibility/cartridge_upgrade_test.lua [hunk 1]
```diff
@@ -7,13 +7,6 @@ local g = t.group()
 
 local helpers = require('test.helper')
 
-local function version(srv)
-    local version = srv:call('require', {'cartridge.VERSION'})
-    local cwd = srv:call('package.loaded.fio.cwd')
-    log.info('Check %s version: %s (from %s)', srv.alias, version, cwd)
-    return version
-end
-
```
Comment (yngvar-antonsson) [inline test/compatibility/cartridge_upgrade_test.lua:16]: Please don't remove this code. You can just change `cartridge.VERSION` -> `cartridge.version`.

**Review by yngvar-antonsson — CHANGES_REQUESTED** [review]

File: test/compatibility/cartridge_upgrade_test.lua [hunk 2]
```diff
@@ -8,7 +8,7 @@ local g = t.group()
 local helpers = require('test.helper')
 
 local function version(srv)
-    local version = srv:call('require', {'cartridge.VERSION'})
+    local version = srv:call('require', {'cartridge.version'})
```
Comment (DifferentialOrange) [inline test/compatibility/cartridge_upgrade_test.lua:11]: I understand you idea to unify the approach, but I think that there is a lot of people who uses `require('cartridge.VERSION')` now. For example, in metrics tests
https://github.com/tarantool/metrics/blob/e0576c226ea0b9c88d05878bc2c0ea991e8889b5/test/helper.lua#L17
(and I think there could be modules which depends on it in its code). I don't know, why it's not a `require('cartridge').VERSION`, but I'm sure we shouldn't rework the file structure for now.

Comment (DifferentialOrange) [reply test/compatibility/cartridge_upgrade_test.lua:11]: Please, restore the file structure in auth-extention too with new PR

https://github.com/tarantool/cartridge-auth-extension/commit/fc53f7bd60942e8038d0c9141d36c19bfb1d1c8a

File: test/compatibility/cartridge_upgrade_test.lua (hunk 1 above)
Comment (GRISHNOV) [reply test/compatibility/cartridge_upgrade_test.lua]: Hmm, as far as I understand, the current check on [scm-1](https://github.com/tarantool/cartridge/actions/runs/4465416213/jobs/7842457371?pr=2056#step:9:24) is no longer [relevant](https://github.com/tarantool/cartridge/actions/runs/4465416213/jobs/7842457371?pr=2056#step:9:24), and the `cartridge_upgrade_test.version` function is also no longer needed

File: test/compatibility/cartridge_upgrade_test.lua (hunk 2 above)
Comment (GRISHNOV) [reply test/compatibility/cartridge_upgrade_test.lua]: Restored, also [opened](https://github.com/tarantool/cartridge-auth-extension/pull/10)

**Comment by DifferentialOrange** [comment]
@GRISHNOV the CI is red, and it is not the flaky tests

**Comment by GRISHNOV** [comment]
> @GRISHNOV the CI is red, and it is not the flaky tests

Yep, it is related with [this](https://github.com/tarantool/cartridge/pull/2056#discussion_r1141684011), AFAIU, the red `scm-1` tests can be deleted like [here](https://github.com/tarantool/cartridge/commit/299a2286ea17111373c2f2261809e52ceccb7602)

File: CHANGELOG.rst [hunk 3]
```diff
@@ -18,6 +18,8 @@ Added
 
 - ``cartridge.roles.metrics`` role (`tarantool#7725 <https://github.com/tarantool/tarantool/issues/7725>`_).
 
+- Versioning support.
```
Comment (yngvar-antonsson) [inline CHANGELOG.rst:21]: Please add a link to an ☂️  issue.

**Comment by yngvar-antonsson** [comment]
> > @GRISHNOV the CI is red, and it is not the flaky tests
> 
> Yep, it is related with [this](https://github.com/tarantool/cartridge/pull/2056#discussion_r1141684011), AFAIU, the red `scm-1` tests can be deleted like [here](https://github.com/tarantool/cartridge/commit/299a2286ea17111373c2f2261809e52ceccb7602)

Ok, then remove It and merge PR after the next CI run.

File: CHANGELOG.rst (hunk 3 above)
Comment (GRISHNOV) [reply CHANGELOG.rst]: Added

File: test/compatibility/cartridge_upgrade_test.lua (hunk 1 above)
Comment (GRISHNOV) [reply test/compatibility/cartridge_upgrade_test.lua:16]: > Ok, then remove It and merge PR after the next CI run.

Done, deleted

**Review by DifferentialOrange — APPROVED** [review]

**Pull request merged**